zephyr: Upgrade to Zephyr v3.1.0.

MaureenHelm opened the pull request:
Updates the Zephyr port build instructions and CI to use the latest
Zephyr release tag.

Tested on frdm_k64f.

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

MaureenHelm committed:
zephyr: Update include paths to use the zephyr namespace.

Zephyr v3.1.0 moved all public headers to include/zephyr.

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

dpgeorge commented:
Thanks for this!

Do you want to wait for v3.1.0 (non-rc version) before merging?

MaureenHelm commented:
> Do you want to wait for v3.1.0 (non-rc version) before merging?

Yes. I posted this as a draft to get a CI run in and catch any potential issues before the Zephyr release is finalized. I'll update the PR and mark it ready for review after the release next week.

MaureenHelm committed:
zephyr: Upgrade to Zephyr v3.1.0.

Updates the Zephyr port build instructions and CI to use the latest
Zephyr release tag.

Tested on frdm_k64f.

Signed-off-by: Maureen Helm <maureen.helm@intel.com>

dpgeorge commented:
Tested on a FRDM K64F, the REPL works.

Rebased and merged in 4fd54a475661953946e9e3c3f962b20619c82240 and bada8b3cada2990d73e6402b029cfb699f3d9438 (I made a small change to the docs to use a valid git hash in the example banner).

dpgeorge commented:
@MaureenHelm I noticed that on the FRDM K64F, after pressing the reset button, it takes about 2 seconds for the MicroPython banner to be printed (after the Zephyr banner is printed, which happens immediately after reset).  Is this "normal"?  Or is MicroPython doing something wrong which takes it a while to start up?

MaureenHelm commented:
> I noticed that on the FRDM K64F, after pressing the reset button, it takes about 2 seconds for the MicroPython banner to be printed (after the Zephyr banner is printed, which happens immediately after reset). Is this "normal"? Or is MicroPython doing something wrong which takes it a while to start up?

The delay is occurring before execution reaches main(), where we start up MicroPython. I suspect it's happening in the Zephyr network stack initialization, because I don't see it with the minimal build configuration where the network stack is disabled.
```
$ west build -b frdm_k64f -- -DCONF_FILE=prj_minimal.conf
```

dpgeorge commented:
I saw this same delay on micro:bit v2 when built using the default settings.  Then I added `bbc_microbit_v2.conf` with some custom settings that had `CONFIG_NETWORKING=n` and the delay went away.  So that could be it.
